src/linux/frida-helper-backend.vala

```diff
@@ -2031,7 +2031,11 @@ namespace Frida {
 				get_regs (&saved_regs);
 				write_memory (target_address, original_code);
 
-				bool hit_breakpoint = saved_regs.program_counter == target_address ||
+				/*
+				 * Note that on ILP32 platforms, these types are 64-bit, but in reality only
+				 * a 32-bit address space is in use.
+				 */
+				bool hit_breakpoint = (void *) saved_regs.program_counter == (void *) target_address ||
```

Comment:
These casts shouldn't be needed unless junk bits have snuck into either side. So ISTM we should rather fix that if so? (This also doesn't change the second part of the condition, which would have the same issue.)